### PR TITLE
feat(overlay,commons): deprecate overlay; migrate references to commons

### PR DESCRIPTION
**Pull request by castastrophe**
## Description

@spectrum-css/overlay and @spectrum-css/commons contain the same overlay extends. This PR aims to deduplicate this work and unify extends under the commons package, deprecating the @spectrum-css/overlay package.

This also migrates to `postcss-extend` instead of `postcss-inherit`. During the build rework, we ran into a lot of compatibility issues using `postcss-inherit` with postcss 8 that were blocking our migration and update.

**@spectrum-css/commons**

 - Moves `commons/overlay-coretokens.css` and `commons/basebutton-coretokens.css` to `commons/*.css` without the `-coretokens.css` postfix because the non-coretokens assets are no longer used in the library.
 - The legacy code for components not yet on coretokens was removed.

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

Components this impacts:

**updates to extend basebutton.css from commons**
- actionbutton
- button
- closebutton
- logicbutton
- picker

**updates to extend overlay.css from commons**
- modal
- popover
- quickaction
- tooltip
- underlay

**other related changes**
- icon (just updates inherit to use extend)
- dialog (removes import because file contains no overlay extends)

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [x] The pages render correctly, are accessible, and are responsive.

2. If components have been modified, VRTs have been run on this branch:

- [x] VRTs have been run and looked at.
- [x] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes.

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] I have updated relevant storybook stories and templates.
- [x] If my change impacts **other components**, I have tested to make sure they don't break.
- [x] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

**Review by mdt2 — APPROVED** [review]
This looks good to me, I tested the code and all the components you listed in Storybook plus a few of them in the docs site.

One question since I'm a bit out of the loop on deprecation process, how are we planning to communicate deprecations like overlay?

**Comment by castastrophe** [comment]
> One question since I'm a bit out of the loop on deprecation process, how are we planning to communicate deprecations like overlay?

Yes, good question.  My plan was to mark the package deprecated in npm. Since it only contains placeholders and no usable CSS, that should be enough to let anyone who might be using it downstream to either migrate off or switch to @spectrum-css/commons.

**Comment by castastrophe** [comment]
`npm deprecate @spectrum-css/overlay "@spectrum-css/overlay is deprecated; all it's placeholders can now be referenced from @spectrum-css/components/commons/overlay.css"`

File: components/commons/basebutton.css [hunk 1]
```diff
@@ -11,7 +11,6 @@ governing permissions and limitations under the License.
 */
```
Comment (castastrophe) [inline components/commons/basebutton.css:11]: Note that this is the basebutton-coretokens.css file that has been moved to basebutton.css and that no changes were made to the actual commons assets.

**Review by jawinn — APPROVED** [review]
This is really helpful in helping to clean up Overlay and removing the duplicate code! Deprecating the component feels like the right way to go.

I think a step 2 for followup refactor work could be to do one of the following:
**a**. remove overlay entirely from commons and bring its necessary CSS directly to the components that need it. `%spectrum-overlay` and `%spectrum-overlay--open` are not providing a lot.
**b**. OR, at the least removing `%spectrum-overlay--bottom--open`, `%spectrum-overlay--top--open`, `%spectrum-overlay--right--open`, `%spectrum-overlay--left--open`, which as you've noted are either not used or only used once.

File: components/page/index.css [hunk 2]
```diff
@@ -10,7 +10,7 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-& {
+:root {
```
[the remark this answers is not in the document]
Comment (castastrophe) [reply components/page/index.css:13]: This is unrelated sorry!  There's a plugin, notnested, that is converting hanging ampersands. I'll revert this from here though.

**Review by pfulton — APPROVED** [review]

**Pull request merged**